Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbDataStore.cs

```diff
@@ -68,14 +68,133 @@ public IEnumerable<string> GetEventsByType(ExecutionEventId eventTypeID)
                 {
                     foreach (var kvp in database.PrefixSearch(eventQuery.ToByteArray()))
                     {
-                        storedEvents.Add(BXLInvocationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                        switch (eventTypeID)

```

Comment:
There is too much repetition going on here.  Can we replace this whole `switch` with something like:
```csharp
if (m_eventParsersDictionary.TryGetValue(eventTypeID, out var parser)
{
    sotredEvents.Add(parser.ParseFrom(kvp.Value).ToString());
}  
```
Then you just need to initialize this `m_eventParsersDictionary` once in the constructor or wherever.

Reply:
Good idea, ill go ahead and implement that.